src/pybind/mgr/dashboard/services/access_control.py

```diff
@@ -181,13 +181,26 @@ def __init__(self, username, password, name=None, email=None, roles=None,
             self.refreshLastUpdate()
         else:
             self.lastUpdate = lastUpdate
+        self.needChangePassword = needChangePassword
```

Comment:
I do have a general question on how to recognize if the password needs to be changed or not.

With regards to http://tracker.ceph.com/issues/40329 which is about a password expiration time-frame, I'm wondering if it would be possible to use `lastPasswordUpdate` field instead of `needChangePassword`. The `lastPasswordUpdate` field would be a time field similar to `lastUpdate`. After the user creation the field can be set to `None` to make sure the user changes the password  at his first login. After that the `lastPasswordUpdate` will be set to the current time.
This would help me with implementing the password expiration feature because I don't need to specify a separate field for that.

It's just an idea. What do you think? Maybe there is an issue with my proposal I currently don't see.

Reply:
Yes, I think it sounds good.